controller/engine_controller.go

```diff
@@ -1814,6 +1832,58 @@ func (ec *EngineController) startRebuilding(e *longhorn.Engine, replica, addr st
 	return nil
 }
 
+// updateReplicaRebuildFailedCondition updates the rebuild failed condition if replica rebuilding failed
+func (ec *EngineController) updateReplicaRebuildFailedCondition(replica *longhorn.Replica, errMsg string) error {
+	replicaRebuildFailedReason, conditionStatus, err := ec.getReplicaRebuildFailedReason(replica.Spec.NodeID, errMsg)
+	if err != nil {
+		return err
+	}
+
+	replica.Status.Conditions = types.SetCondition(
+		replica.Status.Conditions,
+		longhorn.ReplicaConditionTypeRebuildFailed,
+		conditionStatus,
+		replicaRebuildFailedReason,
+		errMsg)
+
+	_, err = ec.ds.UpdateReplicaStatus(replica)
+
```

Comment:
Just noticed, we can get the updated replica from `UpdateReplicaStatus` instead of ignoring it, so we can use the returned replica later.

Reply:
Modified, it will return updated replica. Thanks